app/src/composables/useSiwe.ts

```diff
@@ -124,46 +108,29 @@ export function useSiwe() {
     await executeFetchUserNonce()
     if (!nonce.value) return
 
-    const statement = 'Sign in with Ethereum to the app.'
-    const siweMessageCreator = createSiweMessageCreator(
-      address.value as string,
-      statement,
-      nonce.value.nonce
-    )
-
-    authData.value.message = await siweMessageCreator.create()
+    authData.value.message = createSiweMessage({
+      address: address.value as string,
+      statement: 'Sign in with Ethereum to the app.',
+      nonce: nonce.value.nonce,
+      chainId: chainId.value,
+      uri: window.location.origin,
+      domain: window.location.host,
+      version: '1'
+    })
 
     signMessage({ message: authData.value.message })
   }
 
   async function siwe() {
-    // Check if we have metamask installation befor continue the process
-    const metaMaskConnector = connectors.find(
-      (connector) => connector.name.split(' ')[0] === 'MetaMask'
-    )
-
-    if (!metaMaskConnector) {
-      addErrorToast('MetaMask is not installed, Please install MetaMask to continue')
-      return
-    }
-
     try {
-      isProcessing.value = true
-
-      const networkChainId = parseInt(NETWORK.chainId)
-
-      if (!isConnected.value) {
-        connect({ connector: metaMaskConnector, chainId: networkChainId })
+      if (!(await performChecks())) {
+        isProcessing.value = false
+        return
       }
 
-      if ((await metaMaskConnector.getChainId()) !== networkChainId) {
-        switchChain({
-          chainId: networkChainId,
-          connector: metaMaskConnector
-        })
+      if (address.value) {
+        await executeSiwe()
```

Comment:
You are not throwing an error so no need to have a try catch here. 

Normaly you sould have a ref that represent is the check is good or not. So it should not be the result of `await performChecks()`

await performChecks()
if(address.value && checkStatus){
        await executeSiwe()
}
isProcessing.value = false

// Then the watch will display error depending on what happen.